econml/dowhy.py

```diff
@@ -0,0 +1,222 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+
+"""Helper class to allow other functionalities from dowhy package.
+
+References
+----------
+DoWhy, https://microsoft.github.io/dowhy/
+
+"""
+
+import inspect
+import pandas as pd
+import numpy as np
+from dowhy import CausalModel
+from econml.utilities import check_input_arrays, reshape_arrays_2dim
+
+
+class DoWhyWrapper:
+    """A wrapper class to allow user call other methods from dowhy package through EconML.
+    (e.g. causal graph, refutation test, etc.)
+
+    Parameters
+    ----------
+    cate_estimator: instance
+        An instance of any CATE estimator we currently support
+
+    """
+
+    def __init__(self, cate_estimator):
+        self._cate_estimator = cate_estimator
+
+    def _get_params(self):
+        init = self._cate_estimator.__init__
+        # introspect the constructor arguments to find the model parameters
+        # to represent
+        init_signature = inspect.signature(init)
+        # Consider the constructor parameters excluding 'self'
+        parameters = [p for p in init_signature.parameters.values()
+                      if p.name != 'self' and p.kind != p.VAR_KEYWORD]
+        for p in parameters:
+            if p.kind == p.VAR_POSITIONAL:
+                raise RuntimeError("cate estimators should always "
+                                   "specify their parameters in the signature"
+                                   " of their __init__ (no varargs)."
+                                   " %s with constructor %s doesn't "
+                                   " follow this convention."
+                                   % (self._cate_estimator, init_signature))
```

Comment:
Use a format literal string for readability:

```suggestion
                raise RuntimeError("cate estimators should always "
                                   "specify their parameters in the signature"
                                   " of their __init__ (no varargs)."
                                   f" {self._cate_estimator} with constructor {init_signature} doesn't "
                                   " follow this convention.")
```